src/components/Modals/ConfirmationModal.jsx

```diff
@@ -0,0 +1,107 @@
+// React Imports
+import React from 'react';
+// Material UI Imports
+import Button from '@mui/material/Button';
+import ClearIcon from '@mui/icons-material/Clear';
+import CheckIcon from '@mui/icons-material/Check';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+// Component Imports
+import LogoutButton from './LogoutButton';
+
+/**
+ * confirmationModalProps is an object that stores the props for the
+ * ConfirmationModal component
+ *
+ * @typedef {object} confirmationModalProps
+ * @property {boolean} showConfirmationModal - toggle showing modal
+ * @property {React.Dispatch<React.SetStateAction<boolean>>} setShowConfirmationModal - used to close the modal
+ * @property {string} title - text rendered in dialog title
+ * @property {string} text - text rendered in dialog content text & confirm button
+ * @property {Function} confirmFunction - method that runs onClick of button
+ * @property {boolean} processing - state used to disable button
+ * @property {boolean} [isLogout] - boolean to wrap button with inrupt logout functionality
+ * @memberof typedefs
+ */
+
+/**
+ * ConfirmationModal Component - Component that allows users to cancel or
+ * confirm their previously chosen action
+ *
+ * @memberof Modals
+ * @name ConfirmationModal
+ * @param {confirmationModalProps} props - Props used for ConfirmationModal
+ * @returns {React.JSX.Element} - The confirmation modal
+ */
+const ConfirmationModal = ({
+  showConfirmationModal,
+  setShowConfirmationModal,
+  title,
+  text,
+  confirmFunction,
+  processing,
+  isLogout = false
+}) => {
+  const confirmButton = () =>
```

Comment:
Just thought of this now, but was thinking we could perhaps simplify this further since one is just the same button but with a special wrapper that is `<LogoutButton>`.

I was thinking along the lines of making the `Button` as a component so that it'll become:

```jsx
const AcceptButton = () => (
  <Button
      variant="contained"
      color="primary"
      aria-label="Confirm Button"
      endIcon={<CheckIcon />}
      onClick={confirmFunction}
      disabled={processing}
      sx={{ marginLeft: '1rem' }}
    >
      {title.toUpperCase()}
    </Button>
);

const confirmButton = () => (
  isLogout ? (
    <LogoutButton>
      <AcceptButton />
    </LogoutButton>
  ) : (<AcceptButton />)
);
```

Reply:
Refactored into it's own component `ConfirmationButton`.

Reply:
@leekahung see what you think, and I fixed conflict.

Reply:
Think it looks good! Thanks for the updates.